src/bp/ui-studio/src/web/views/OneFlow/diagram/VariableTypesForm/EnumForm.tsx

```diff
@@ -75,6 +78,28 @@ const EnumForm: FC<Props> = ({
     updateFormItem(convertFromTags(data))
   }
 
+  const isDupplicate = (localItem: Item[], newItem: Item) => {
+    const lastAdded: string = [newItem.name].concat(newItem.tags).slice(-1)[0]
+    for (const occurence of localItem) {
+      if ([occurence.name].concat(occurence.tags).includes(lastAdded)) {
+        toastFailure(`${lastAdded} already exists in ${occurence.name}`)
+        return false
+      }
+    }
+
+    const localItemNames = localItem.map(i => i.name)
+    for (const entity of allEntities) {
+      const noLocalEntity = entity.occurrences.filter(o => !localItemNames.includes(o.name))
```

Comment:
Not sure to understand what happens here. You already check if it's a duplicate in the current entity, why not just exclude the current entity from allEntities ?

Reply:
it's because allEntities contains all the entities of all enumerations, they are updated when the form lose focus.

 localItem contains only the current enumeration which is updated more frequently. 

newItem is the newest updated component which is then used to update localItem but we need to check for duplicate first.